rust/gelato/src/chains.rs

```diff
@@ -0,0 +1,151 @@
+// Ideally we would avoid duplicating ethers::types::Chain, but we have enough need to justify
+// a separate, more complete type conversion setup, including some helpers for e.g. locating
+// Gelato's verifying contracts. Converting from the chain's name in string format is useful
+// for CLI usage so that we don't have to remember chain IDs and can instead refer to names.
+
+use ethers::types::{Address, U256};
+use std::{fmt, str::FromStr};
+
+use crate::err::GelatoError;
+
+// This list is currently trimmed to the *intersection* of
+// {chains used by Abacus in any environment} and {chains included in ethers::types::Chain}.
+// Notably missing is Celo/Alfajores.
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum Chain {
+    Mainnet = 1,
+    Rinkeby = 4,
+    Goerli = 5,
+    Kovan = 42,
+    Polygon = 137,
+    PolygonMumbai = 80001,
+    Avalanche = 43114,
+    AvalancheFuji = 43113,
+    Arbitrum = 42161,
+    ArbitrumTestnet = 421611,
+    Optimism = 10,
+    OptimismKovan = 69,
+    BinanceSmartChain = 56,
+    BinanceSmartChainTestnet = 97,
+}
+
+impl fmt::Display for Chain {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "{:?}", self)
+    }
+}
+
+impl FromStr for Chain {
+    type Err = GelatoError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
```

Comment:
You might like to know that you can `use Chain::*` within the function and avoid having to type as much. I don't always do this, but sometimes it is nice.

Reply:
I didn't know, thanks, that's handy. As above, by default I'd just use the same convention as the ethers chains impl in https://github.com/gakonst/ethers-rs/blob/master/ethers-core/src/types/chain.rs if you don't have a strenuous objection

Reply:
I did approve the PR. Leaving it is fine.